core/src/test/scala/unit/kafka/log/UnifiedLogTest.scala

```diff
@@ -3464,13 +3474,13 @@ class UnifiedLogTest {
       new SimpleRecord("c".getBytes)), 5)
 
 
-    log.updateHighWatermark(2L)
+    log.updateHighWatermark(3L)
```

Comment:
This was the test I was referring to in my above comment. I changed this to 3 to align onto a batch boundary, but I'm not sure about the case if the HWM is not batch aligned.